lib/symboldatabase.cpp

```diff
@@ -4175,13 +4175,24 @@ void Function::addArguments(const SymbolDatabase *symbolDatabase, const Scope *s
         }
     }
 
+    auto skipLambda = [](const Token* tok) -> const Token* {
```

Comment:
Whats the difference between `skipLambda` and `findLambdaEndToken`?

Reply:
`findLambdaEndToken()` requires the AST, which an inner lambda (e.g. in the parameter list) is not part of. Failure of `findLambdaEndToken()` then generates an InternalError, which is what #10831 is about.

Reply:
I see then maybe we should name it `findLambdaEndTokenWithoutAST`. It also would be better to make this a static function instead of a lambda as well.